packages/lib-utils/src/components/details-page-header/utils/ActionMenu.tsx

```diff
@@ -0,0 +1,168 @@
+
+export type ActionCTA =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | { callback: (event: React.MouseEvent<any> | React.KeyboardEvent | MouseEvent) => void }
+  | { href: string; external?: boolean };
+
+export type Action = {
+  /** A unique identifier for this action. */
+  id: string;
+  /** The label to display in the UI. */
+  label: React.ReactNode;
+  /** Subtext for the menu item */
+  description?: string;
+  /** Executable callback or href.
+   * External links should automatically provide an external link icon on action.
+   * */
+  cta: ActionCTA;
+  /** Whether the action is disabled. */
+  isDisabled?: boolean;
+  /** The tooltip for this action. */
+  tooltip?: string;
+  /** The icon for this action. */
+  icon?: React.ReactNode;
+};
+
+export type GroupedActions = {
+  /** A unique identifier for this group. */
+  groupId: string;
+  /** Optional label to display as group heading */
+  groupLabel?: string;
+  /** Actions under this group. */
+  groupActions: Action[];
+};
+
+export enum ActionMenuVariant {
+  KEBAB = 'plain',
+  DROPDOWN = 'default',
+}
+
+export type ActionMenuProps = EitherNotBoth<
+  { actions: Action[] },
+  { groupedActions: GroupedActions[] }
+> & {
+  /** Optional flag to indicate whether action menu should be disabled */
+  isDisabled?: boolean;
+  /** Optional variant for action menu: DROPDOWN vs KEBAB (defaults to dropdown) */
+  variant?: ActionMenuVariant;
+  /** Optional label for action menu (defaults to 'Actions') */
+  label?: string;
+  /** Optional position (left/right) at which the action menu appears (defaults to right) */
+  position?: DropdownPosition;
+};
+
+export const ActionMenu: React.FC<ActionMenuProps> = ({
+  actions = [],
+  groupedActions = [],
+  isDisabled,
+  variant = ActionMenuVariant.DROPDOWN,
+  label = 'Actions',
+  position = DropdownPosition.right,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isGrouped, setIsGrouped] = React.useState(false);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [dropdownActionItems, setDropdownActionItems] = React.useState<any[]>([]);
+
+  React.useEffect(() => {
+    if (!_.isEmpty(groupedActions)) {
+      setIsGrouped(true);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const onToggle = (open: boolean) => {
+    setIsOpen(open);
+  };
+
+  const onFocus = () => {
+    const element = document.getElementById(`toggle-menu-${label}`);
+    if (element) {
+      element.focus();
+    }
+  };
+
+  const onSelect = () => {
+    setIsOpen(false);
+    onFocus();
+  };
+
+  /** Returns a DropDownItem element corresponding to an action */
+  const dropdownActionItem = React.useCallback((action: Action) => {
+    const externalIcon =
+      'href' in action.cta && 'external' in action.cta && action.cta.href && action.cta.external ? (
+        <ExternalLinkAltIcon />
+      ) : null;
+    const icon = action.icon ?? externalIcon;
```

Comment:
By using the DropdownItem icon field for the ExternalLink icon, it shows before the text. Typically we've show it after the text.

Reply:
@jeff-phillips-18 Thanks! Are you referring to usages in the OCP Console repo? Could you point me to an example please?

Reply:
![image](https://user-images.githubusercontent.com/11633780/191488876-aaf1ad93-052f-4234-ad49-2d83b8888ab3.png)

![image](https://user-images.githubusercontent.com/11633780/191489203-9feffd6f-1b79-48cc-9c9d-630dbbd3f037.png)

Reply:
@jeff-phillips-18 I've pushed up a commit with an option to specify the icon after text. Since we're using a DropdownItem I've maintained the default configuration as icon appearing first, but it can be customized.